File: Frontend/src/app/validator.ts
import {Injectable} from "@angular/core";

@Injectable()
export class Validator {
  public validirajEmail(email: string) {
    let regex: RegExp = RegExp('^[\\w-\.]+@([\\w-]+\.)+[\\w-]{2,4}$');
    return regex.test(email);
  }

  public validirajLozinku(lozinka: string) {
    let regex = RegExp('^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$')
    return regex.test(lozinka);
  }

  public validirajText(text: string) {
    let regex = RegExp('^[A-Za-z]+$')
    return regex.test(text);
  }
}

File: HCI-Project/Backend/Servisi/Validator.cs
using System.Text.RegularExpressions;

namespace Backend.Servisi;

public class Validator
{
    public bool ValidirajEmail(string email)
    {
        var regex = new Regex("[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}");
        return regex.IsMatch(email);
    }

    public bool ValidirajLozinku(string lozinka)
    {
        var regex = new Regex("^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$");
        return regex.IsMatch(lozinka);
    }

    public bool ValidirajText(string text)
    {
        var regex = new Regex("[A-Za-z]+");
        return regex.IsMatch(text);
    }
}